core/modules/layout/tests/layout.test

```diff
@@ -1133,14 +1133,22 @@ class LayoutInterfaceTest extends BackdropWebTestCase {
     );
     $this->backdropPost('admin/structure/layouts/manage/' . $layout_name . '/configure', $edit, t('Save layout'));
 
-    $deleted_uuids = $default_layout->positions['sidebar'];
+    $orphaned_uuids = $default_layout->positions['sidebar'];
     layout_reset_caches();
     $layout = layout_load($layout_name);
-    $this->assertBlocksMatch($layout, $default_layout);
-    $this->assertFalse(isset($layout->positions['sidebar']), 'Sidebar region blocks removed from layout when changing layout to single column.');
+    $this->assertBlocksMatch($layout, $default_layout, $orphaned_uuids);
+    $this->assertFalse(isset($layout->positions['sidebar']), 'Sidebar region blocks moved out when changing layout to single column.');
 
-    $intersection = array_intersect(array_keys($layout->content), $deleted_uuids);
+    $intersection = array_intersect(array_keys($layout->content), $orphaned_uuids);
     $this->assertEqual($intersection, array(), 'The sidebar blocks have been removed from the layout content array.');
+    $default_layout_sidebar_content_regions_count = count($default_layout->positions['sidebar']) + count($default_layout->positions['content']);
+    $new_content_region_count = count($layout->positions['content']);
+    // Count the system main block separately since it doesn't exist in the
+    // new layout.
+    if ($default_layout->positions['content'][0] == 'default') {
+      $new_content_region_count++;
+    }
+    $this->assertTrue($default_layout_sidebar_content_regions_count == $new_content_region_count, 'Sidebar region blocks have been moved to the default region.');
```

Comment:
I think it may be safer to actually check that the actual blocks in the default layout sidebar match the blocks in the new layout content area rather than testing a count. A change in the future that messes with the blocks but doesnt change the number of blocks would slip through this.

Reply:
Finding the "actual" blocks seems to be quite hard. The existing assertBlocksMatch only checks if the block's module and delta match, not if they're the same exactly. In our case the block's module should be the same but the delta won't be. I could try write something specific to this one test, but there's no point right now to make it more generic since we only need to check an orphaned block once. At least for now.